lumicks/pylake/channel.py

```diff
@@ -21,14 +22,54 @@ class Slice:
         Plot labels: "x", "y", "title".
     calibration: ForceCalibration
     """
-    def __init__(self, data_source, labels=None, calibration=None):
+    def __init__(self, data_source, labels=None, calibration=None, unit=None):
         self._src = data_source
         self.labels = labels or {}
+        self.unit = unit
         self._calibration = calibration
 
     def __len__(self):
         return len(self._src)
 
+    def _unpack_other(self, other):
+        if isinstance(other, self.__class__):
+            return other._src.data
+        else:
+            return other
+
+    def _determine_unit(self, other, operation):
+        other_unit = getattr(other, "unit", None)
+        if other_unit:
+            return determine_unit(self.unit, other_unit, operation)
```

Comment:
This function has a path without a `return`. Python does an implicit `return None` in that case, but we really should be explicit if at least one path has a `return`.

Reply:
Actually, this comment made me think.

It's probably wiser to be more explicit and move lack of unit handling into units. For addition and subtraction, adding a value results in the unit that is present; because this is the only operation that makes any sense. For the other operations, it results in an undefined unit (signified by None).

It does mean that when working with datatypes that have no explicit unit, strict unit checking goes out the window (but I don't think that can be avoided, really).